beastling/models/basemodel.py

```diff
@@ -492,7 +492,7 @@ def add_likelihood(self, likelihood):
         dataset.
         """
         for n, f in enumerate(self.features):
-            fname = "%s:%s" % (self.name, f)
+            fname = "%s:%s" % (self.name, xml.valid_id(f))
```

Comment:
Somewhat accidentally I came across the issue with commas in feature names leading to invalid BEAST XML. There might be a fix for this somewhere else, but the implementation here simply uses a small helper function `xml.valid_id` to clean IDs for `id` and `idref` attributes.

Reply:
Rather than having to remember to make calls to `xml.valid_id` in dozens of scattered places, does it make sense to have a method called before `validate_ids` which just runs through the XML document and applies this helper to anything that looks like an ID?

Not asking you to do this for this PR, I'm happy to do it, just want to make sure I'm not overlooking some problem with this approach...

Reply:
I'm already calling `xml.valid_id` for each `id` or `idref` passed into any of the `xml.*` element creation functions. The reason it's still called explicitly here is that `fname` is used a couple of times further down in that method, to create attributes which would not be converted automatically.

Reply:
See https://github.com/lmaurits/BEASTling/pull/247/files#diff-39e48eb1d171082b5a1bacd3f8fe38a2R18-R22